gettsim/pre_processing/piecewise_functions.py

```diff
@@ -1,45 +1,84 @@
+import copy
+
 import numpy as np
 
+from gettsim.pre_processing.generic_functions import check_threholds
+
 
-def piecewise_linear(
-    value,
+def piecewise_polynomial(
+    x,
     lower_thresholds,
     upper_thresholds,
     rates,
     intercepts_at_lower_thresholds,
     rates_modified=False,
 ):
-    """Return a fraction of *value* defined by a piecewise linear function.
+    """Calculate value of the piecewise function at x. The function is defined by the
+    other input arrays.
 
     Args:
-        value (float, >0): The value that the function is applied to.
-        lower_thresholds (1-d float): The lower thresholds defining the intervals
-        upper_thresholds (1-d float): The upper thresholds defining the intervals
-        rates (1-d float): The slope in the interval below the corresponding element
-            of *upper_thresholds*
-        intercepts_at_lower_thresholds (1-d float): the fraction piecewise_linear
-            calculates at the respective thresholds
+        x (float, >0): The value that the function is applied to.
+        lower_thresholds (1-d array): The lower thresholds of each interval.
+        upper_thresholds (1-d array): The upper thresholds each interval.
+        rates (n-d arry): The slope in each interval. Where n is the degree of the
+        polynomial function.
+        intercepts_at_lower_thresholds (1-d array): The intercepts at the lower
+        threshold of each interval.
         rates_modified: Boolean variable indicating, that intercepts can't be used
         anymore.
```

Comment:
used anymore => intercepts from the param file cannot be used (?)

Reply:
This is for the case that we have an individual piecewise linear function. I implemented it for `eink_anr_frei` in 2005. There the rates are multiplied by the nettoquote and we can't use the intercepts.